test/test_encryption.py

```diff
@@ -1942,6 +1944,44 @@ def test_via_loading_shared_library(self):
         with self.assertRaises(ServerSelectionTimeoutError):
             no_mongocryptd_client.db.command("ping")
 
+    # https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/README.rst#20-bypass-creating-mongocryptd-client-when-shared-library-is-loaded
+    @unittest.skipUnless(os.environ.get("TEST_CRYPT_SHARED"), "crypt_shared lib is not installed")
+    def test_client_via_loading_shared_library(self):
+        class TCPHandler(socketserver.BaseRequestHandler):
+            def handle(self):
+                assert not self.request.recv(1024)
```

Comment:
We don't even need the recv here. If this method ever gets called at all then it means a connection to the mongocryptd port was established and the test must fail. Also can you confirm the behavior of raising an exception here? According to the docs:
> If the user-provided [handle()](https://docs.python.org/3/library/socketserver.html#socketserver.BaseRequestHandler.handle) method of the handler class raises an exception, the server’s [handle_error()](https://docs.python.org/3/library/socketserver.html#socketserver.BaseServer.handle_error) method will be called.
> ... The default action is to print the traceback to standard error and continue handling further requests.

Reply:
This is the error when the address is being used:
```
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 980, in _bootstrap_inner
    self.run()
  File "/Users/julius/Work/mongo-python-driver/test/utils.py", line 983, in run
    super(ExceptionCatchingThread, self).run()
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 917, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/julius/Work/mongo-python-driver/test/test_encryption.py", line 1956, in listen
    with socketserver.TCPServer(addr, TCPHandler) as sock:
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/socketserver.py", line 452, in __init__
    self.server_bind()
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/socketserver.py", line 466, in server_bind
    self.socket.bind(self.server_address)
OSError: [Errno 48] Address already in use
```
This is when the address times out:
```
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 980, in _bootstrap_inner
    self.run()
  File "/Users/julius/Work/mongo-python-driver/test/utils.py", line 983, in run
    super(ExceptionCatchingThread, self).run()
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 917, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/julius/Work/mongo-python-driver/test/test_encryption.py", line 1957, in listen
    sock.server_bind()
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/socketserver.py", line 466, in server_bind
    self.socket.bind(self.server_address)
OSError: [Errno 22] Invalid argument
```
I added an assertion to make this distinction clear.